arduino-ide-extension/src/browser/serial/monitor/serial-monitor-send-input.tsx

```diff
@@ -6,6 +6,75 @@ import { BoardsServiceProvider } from '../../boards/boards-service-provider';
 import { MonitorModel } from '../../monitor-model';
 import { Unknown } from '../../../common/nls';
 
+class RingList {
+  protected ring: string[];
+  protected size: number;
+  protected begin: number;
+  protected index: number;
+  protected end: number;
+
+  constructor(size: number = 100) {
+    this.Init = this.Init.bind(this);
+    this.Push = this.Push.bind(this);
+    this.Prev = this.Prev.bind(this);
+    this.Next = this.Next.bind(this);
+    this.Init(size);
+  }
+
+  public Init(size: number = 100)
```

Comment:
You `Init` the `RingList` from the `constructor` function; please change the visibility to `private`.

Reply:
Done